packages/edit-site/src/components/editor-canvas-container/index.js

```diff
@@ -90,7 +90,7 @@ function EditorCanvasContainer( {
 			onClose();
 		}
 		setIsListViewOpened( showListViewByDefault );
-		setEditorCanvasContainerView( undefined );
+		setEditorCanvasContainerView( null );
```

Comment:
Note that changing all reducers to use `undefined` instead of a `null` is likely to end up being a regression, since some of the reducers might make a strict differentiation between the `null` and `undefined`. This is especially valid for custom `@wordpress/data` stores that are already using `undefined` as a valid value somewhere in their reducer trees.

With that in mind, I think we might have to consider adding our own wrapper around the Redux `combineReducers` that handles `undefined` in a way that allows supporting it as a valid value.

Reply:
@jsnajdr what are your thoughts on the above comment?

Reply:
Now, after I abandoned the idea to use `combineReducers` from Redux and created my own version instead, `undefined` state is no longer a concern. And neither are extra keys in state. `@wordpress/data` reducers can continue to use them without any change.

However, this PR still contains changes to Gutenberg's data stores that remove `undefined` state and replace it with `null`. They're no longer strictly needed, but still seem to me like a good code quality improvement.

But these changes are not worth doing if they introduce any backward compatibility issues for the existing Gutenberg stores. But are there any? And if there are, can they be solved?

For example, the `setEditorCanvasContainerView( undefined )` call no longer works and has to be changed to `setEditorCanvasContainerView( null )`. But that can be easily avoided by defaulting the action parameter to `null`: `( view = null ) => ...`. I haven't 100% verified this yet, I only know that unit and e2e tests are green.